OT Primer Knowledge Check Questions (#66)
openthread/ot-docs: site/en/guides/thread-primer/node-roles-and-types.md

*   A Thread network might be composed of multiple partitions

## Check your understanding

<div>
  <devsite-multiple-choice>
    <div>A Thread network device can fulfill one of two forwarding roles. 
    What are they?</div>
    <div>
      <div>Child Node.</div>
      <div>Incorrect.</div>
    </div>
    <div correct>
      <div>Router.</div>
      <div>Correct.</div>
    </div>
    <div correct>
      <div>End Device.</div>
      <div>Correct.</div>
    </div>
    <div>
      <div>Gateway.</div>
      <div>Incorrect.</div>
    </div>
  </devsite-multiple-choice>
</div>

<div>
  <devsite-multiple-choice>
    <div>What are the two primary types of Thread device?</div>
    <div correct>
      <div>Minimal Thread Device (MTD).</div>
      <div>Correct.</div>
    </div>
    <div correct>
      <div>Full Thread Device (FTD).</div>
      <div>Correct.</div>
    </div>
    <div>
      <div>Miniscule Thread Device (MTD).</div>
      <div>Incorrect.</div>
    </div>
    <div>
      <div>Sleepy End Device (SED).</div>
      <div>Incorrect.</div>
    </div>
  </devsite-multiple-choice>
</div>

<div>
  <devsite-multiple-choice>

>     <div>Which of the following statements about Routers are incorrect?</div>

Reviewer: I found it confusing to ask what's incorrect and have wrong answers say correct in the revealed text.

Can we say "Which of the following statements about Routers is not true?" and instead of "Correct" in the answer reveals say "This is true."

Author: Good catch - my oversight. And I like your suggestion.

      <div>That's right. Under these circumstances, a REED can promote itself
        to a Router.</div>
    </div>
    <div>
      <div>When it is an End Device seeking to join the Thread network.</div>
      <div>Incorrect.</div>
    </div>
    <div>
      <div>When it is a REED and the Thread network has merged with a larger
        network.</div>
      <div>Incorrect.</div>
    </div>
  </devsite-multiple-choice>
</div>

<div>
  <devsite-multiple-choice>
    <div>When can a Router cause itself to stop acting as a Router?</div>
    <div correct>
      <div>When it has no children.</div>
      <div>That's correct. A Router with no children is allowed to revert to
        an End Device on its own.</div>
    </div>
    <div>
      <div>When a new End Device is seeking to join the
        Thread network.</div>
      <div>Wrong. A Router cannot revert to an End Device in this scenario.
      </div>
    </div>
    <div>
      <div>When another device on the network elects to become a Router.
      </div>
      <div>Wrong.
      </div>
    </div>
  </devsite-multiple-choice>
</div>

<div>
  <devsite-multiple-choice>
    <div>Consider a scenario where a Thread network contains two 

>     groups of nodes that have radio connectivity with one another, but not

Reviewer: ...two groups of nodes that each have radio connectivity within a group, but not...

    with members of the other group.
    What conclusion can one draw from this?</div>
    <div correct>
        <div>There is more than one partition in that network.</div>
        <div>Correct. A partition is formed around each group
        of nodes that is able to communicate with one another. If there are
        multiple groups of nodes that can communicate amongst themselves, but
        not with members of other groups, then you know that these groups
        each constitute a distinct partition.
        </div>
    </div>
    <div>
      <div>The network lost its Leader.</div>
      <div>Incorrect.</div>
    </div>
    <div>
      <div>All the Routers in the network have gone offline.</div>
      <div>Incorrect. In that case, none of the nodes would be able to
      communicate with one another.</div>
    </div>    
  </devsite-multiple-choice>
</div>

<div>
  <devsite-multiple-choice>
    <div>What device is used to connect Thread and non-Thread networks?</div>
    <div>

>       <div>a Gateway</div>

Reviewer: Capitalize the "a" or drop it for each answer

      <div>Although the term 'Gateway', in traditional networking, does refer
      to a device that connects two networks, there is a more accurate term.
      </div>
    </div>
    <div correct>
      <div>a Border Router</div>
      <div>Correct. A Border Router is used to connect Thread and non-Thread
        networks.</div>
    </div>
    <div>
      <div>a Firewall</div>
      <div>That is not right.</div>
    </div>
    <div>
      <div>a Bridge</div>
      <div>Incorrect. This term refers to a similar concept in traditional
        networking referring to a device that connects two LANs that use the
        same network protocol.</div>
    </div>
  </devsite-multiple-choice>
</div>

<div>
  <devsite-multiple-choice>
    <div>How many Leaders may a Thread network have?</div>
    <div>
      <div>Either none or exactly one.</div>
      <div>This is wrong. A Thread network cannot have fewer than one Leader.
      </div>
    </div>
    <div correct>
      <div>One and only one.</div>
      <div>Correct. A Thread network may have one and only one Leader.</div>
    </div>
    <div>
      <div>More than one.</div>

>       <div>Wrong; a Thread network cannot have multiple Leaders.</div>

Reviewer: No need for a semi-colon here, inconsistent with other answers.

    </div>
  </devsite-multiple-choice>
</div>

## License

> Copyright (c) 2022, The OpenThread Authors.

Reviewer: I think the copyright year for all the pages you're updating should be 2021-22, @jwhui can you confirm?

      a device must keep its transceiver online at all times.</div>
    </div>
    <div>
      <div>A Router provides secure commissioning services for devices trying 
      to join the network.</div>
      <div>This statement is true. Commissioning is an important function of a
      Thread Router.</div>
    </div>
  </devsite-multiple-choice>
</div>

<div>
  <devsite-multiple-choice>
    <div>When can a device upgrade itself to a Router?</div>
    <div correct>
      <div>When it is a REED and it is the only node in reach of a new End
        Device seeking to join the Thread network.</div>
      <div>That's right. Under these circumstances, a REED can promote itself
        to a Router.</div>
    </div>
    <div>
      <div>When it is an End Device seeking to join the Thread network.</div>
      <div>Incorrect.</div>
    </div>
    <div>
      <div>When it is a REED and the Thread network has merged with a larger
        network.</div>
      <div>Incorrect.</div>
    </div>
  </devsite-multiple-choice>
</div>

<div>
  <devsite-multiple-choice>
    <div>When can a Router cause itself to stop acting as a Router?</div>
    <div correct>
      <div>When it has no children.</div>

>       <div>That's correct. A Router with no children is allowed to revert to

Reviewer 2: ```suggestion
      <div>That's correct. A Router with no children may revert to
```
There are other conditions that need to be satisfied (e.g. it does not cause a partition in the Thread network).

        an End Device on its own.</div>
    </div>
    <div>
      <div>When a new End Device is seeking to join the
        Thread network.</div>
      <div>Wrong. A Router cannot revert to an End Device in this scenario.
      </div>
    </div>
    <div>

>       <div>When another device on the network elects to become a Router.

Reviewer 2: This could be true. If the number of Thread routers increases to 24 or more, existing Thread routers can start evaluating whether to become an end device.
